compiler/qsc_frontend/src/resolve/tests.rs

```diff
@@ -2033,6 +2040,263 @@ fn dropped_callable() {
+fn resolved_not_higher_level_callable_from_base() {
+    check_with_capabilities(
+        indoc! {"
+            namespace A {
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function B() : Unit {
+                    Dropped();
+                }
+            }
+        "},
+        TargetCapabilityFlags::empty(),
+        &expect![[r#"
+            namespace namespace7 {
+                @Config(not HigherLevelConstructs)
+                function item1() : Unit {}
+
+                function item2() : Unit {
+                    item1();
+                }
+            }
+        "#]],
+    );
+}
+
+#[test]
+fn dropped_not_higher_level_and_adaptive_callable_from_base() {
+    check_with_capabilities(
+        indoc! {"
+            namespace A {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function B() : Unit {
+                    Dropped();
+                }
+            }
+        "},
+        TargetCapabilityFlags::empty(),
+        &expect![[r#"
+            namespace namespace7 {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function item1() : Unit {
+                    Dropped();
+                }
+            }
+
+            // NotAvailable("Dropped", "A.Dropped", Span { lo: 143, hi: 150 })
+        "#]],
+    );
+}
+
+#[test]
+fn dropped_not_higher_level_and_adaptive_callable_from_unrestricted() {
+    check_with_capabilities(
+        indoc! {"
+            namespace A {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function B() : Unit {
+                    Dropped();
+                }
+            }
+        "},
+        TargetCapabilityFlags::all(),
+        &expect![[r#"
+            namespace namespace7 {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function item1() : Unit {
+                    Dropped();
+                }
+            }
+
+            // NotAvailable("Dropped", "A.Dropped", Span { lo: 143, hi: 150 })
+        "#]],
+    );
+}
+
+#[test]
+fn resolved_not_higher_level_and_adaptive_callable_from_adaptive() {
+    check_with_capabilities(
+        indoc! {"
+            namespace A {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
+                function Dropped() : Unit {}
+
+                function B() : Unit {
+                    Dropped();
+                }
+            }
+        "},
+        TargetCapabilityFlags::Adaptive,
+        &expect![[r#"
+            namespace namespace7 {
+                @Config(Adaptive)
+                @Config(not HigherLevelConstructs)
```

Comment:
Is it possible to test from something other than HigherLevelConstructs?

Reply:
Good idea, I'll add some of those.